rust/processor/src/db/common/models/default_models/parquet_write_set_changes.rs

```diff
@@ -58,98 +58,117 @@ impl Default for WriteSetChange {
-                    block_timestamp,
-                )),
-            ),
-            WriteSetChangeEnum::DeleteResource(inner) => (
-                Self {
-                    txn_version,
-                    state_key_hash: standardize_address(
-                        hex::encode(inner.state_key_hash.as_slice()).as_str(),
-                    ),
-                    block_height,
-                    change_type,
-                    resource_address: standardize_address(&inner.address.to_string()),
-                    write_set_change_index,
-                    block_timestamp,
-                },
-                WriteSetChangeDetail::Resource(MoveResource::from_delete_resource(
+            WriteSetChangeEnum::WriteModule(inner) => {
+                return Ok(Some((
+                    Self {
+                        txn_version,
+                        state_key_hash: standardize_address(
+                            hex::encode(inner.state_key_hash.as_slice()).as_str(),
+                        ),
+                        block_height,
+                        change_type,
+                        resource_address: standardize_address(&inner.address.to_string()),
+                        write_set_change_index,
+                        block_timestamp,
+                    },
+                    WriteSetChangeDetail::Module(MoveModule::from_write_module(
+                        inner,
+                        write_set_change_index,
+                        txn_version,
+                        block_height,
+                    )),
+                )));
+            },
+            WriteSetChangeEnum::DeleteModule(inner) => {
+                return Ok(Some((
+                    Self {
+                        txn_version,
+                        state_key_hash: standardize_address(
+                            hex::encode(inner.state_key_hash.as_slice()).as_str(),
+                        ),
+                        block_height,
+                        change_type,
+                        resource_address: standardize_address(&inner.address.to_string()),
+                        write_set_change_index,
+                        block_timestamp,
+                    },
+                    WriteSetChangeDetail::Module(MoveModule::from_delete_module(
+                        inner,
+                        write_set_change_index,
+                        txn_version,
+                        block_height,
+                    )),
+                )));
+            },
+            WriteSetChangeEnum::WriteResource(inner) => {
+                let resource_option = MoveResource::from_write_resource(
                     inner,
                     write_set_change_index,
                     txn_version,
                     block_height,
                     block_timestamp,
-                )),
-            ),
+                );
+
+                if let Some(resource) = resource_option.unwrap() {
```

Comment:
Are we confident this will never fail? If so, can we use `expect` with a message that explains why, plus a comment? Unless we're 100% certain, it'd be better to just `.context("explanation")?` and bubble up the error, since this function also returns a Result.

Reply:
I think if there isn't, that's a problematic txn. and it seems like we have been panicing if there is a problematic one as I see expect in the default model. (move_resource.rs, and write_set_changes.rs). 

but if it makes sense to use `context`  here as it already returns a Result.